src/IonParserTextRaw.ts

```diff
@@ -1060,9 +1062,13 @@ export class ParserTextRaw {
                             index += this._esc_len;
                         }
                         if (this.isLowSurrogate(tempChar)) {
-                            s += ch + tempChar;
+                            // convert from UTF-16 surrogate pair to a codepoint
+                            let hiSurrogate = ch;
+                            let loSurrogate = tempChar;
+                            let codepoint = 0x10000 + ((hiSurrogate & _UTF16_MASK) << 10) + (loSurrogate & _UTF16_MASK);
+                            s += String.fromCodePoint(codepoint);
                         } else {
-                            throw new Error("illegal high surrogate" + ch);
+                            throw new Error("illegal low surrogate: " + ch);
```

Comment:
shouldn't this still say high surrogate?

Reply:
This error is thrown in the else block of an `isLowSurrogate?` test, so I think "illegal low surrogate" is the right error message.  The code was expecting to find a low surrogate, but didn't.

Reply:
means an unpaired high surrogate

Reply:
I've attempted to clarify this as well as the "illegal low surrogate" error thrown a few lines below this.  Feel free to suggest better messages.